### PR TITLE
Migrated sale_order_general_discount to v13.

### PR DESCRIPTION
**Pull request by sudhir-erpharbor**

### CODE REVIEW

**Comment by sergio-teruel** [comment]
@sudhir-erpharbor Are you going to continue this PR?

File: sale_order_general_discount/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Copyright 2018 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Sale Order General Discount",
+    "summary": "General discount per sale order",
+    "version": "13.0.1.0.0",
+    "development_status": "Production/Stable",
```
Comment (sergio-teruel) [inline sale_order_general_discount/__manifest__.py:7]: Add maintainers key

File: sale_order_general_discount/models/sale.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+from odoo.addons import decimal_precision as dp
+from lxml import etree
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    general_discount = fields.Float(
+        digits=dp.get_precision("Discount"), string="Discount (%)"
+    )
```
Comment (sergio-teruel) [inline sale_order_general_discount/models/sale.py:13]: Make field compute, store, and readonly False... to change to new onchange behavior

File: sale_order_general_discount/models/sale.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    @api.onchange("partner_id")
```
Comment (sergio-teruel) [inline sale_order_general_discount/models/sale.py]:
```suggestion
    @api.depends("partner_id")
    def _compute_general_discount(self):
        for so in self:
            so.general_discount = so.partner_id.sale_discount
```

File: sale_order_general_discount/models/sale.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    def onchange_partner_id(self):
+        super().onchange_partner_id()
+        self.general_discount = self.partner_id.sale_discount
+        return
+
+    @api.onchange("general_discount")
```
Comment (sergio-teruel) [inline sale_order_general_discount/models/sale.py:20]: make the same....

File: sale_order_general_discount/tests/test_sale_order_general_discount.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+                "product_uom_qty": 1,
+                "product_uom": cls.product.uom_id.id,
+                "price_unit": 1000.00,
+            })],
+            "pricelist_id": cls.env.ref("product.list0").id,
+        })
+        cls.View = cls.env["ir.ui.view"]
+
+    def test_default_partner_discount(self):
+        self.order.onchange_partner_id()
```
Comment (sergio-teruel) [inline sale_order_general_discount/tests/test_sale_order_general_discount.py]: With changes commented is not needed

**Review by sergio-teruel — CHANGES_REQUESTED** [review]

**Comment by sudhir-erpharbor** [comment]
> @sudhir-erpharbor Are you going to continue this PR?

Yes, will make the changes as per your request.

**Comment by sergio-teruel** [comment]
Ok... Thanks... :smile:

File: sale_order_general_discount/models/sale.py [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+from odoo import api, fields, models
+from odoo.addons import decimal_precision as dp
+from lxml import etree
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    general_discount = fields.Float(
+        digits=dp.get_precision("Discount"), string="Discount (%)"
```
Comment (sergio-teruel) [inline sale_order_general_discount/models/sale.py]: get_precision method is deprecated... 
See --> https://github.com/odoo/odoo/blob/1a315e87b05fcbdefcb3db6d8d4837f6cc3e835c/odoo/addons/base/models/decimal_precision.py#L13

File: sale_order_general_discount/__manifest__.py (hunk 1 above)
Comment (sudhir-erpharbor) [reply sale_order_general_discount/__manifest__.py:7]: @sergio-teruel What should I add in "maintainers"? Just my github user name or anything else?

**Comment by rousseldenis** [comment]
@sudhir-erpharbor Do you plan to finish this or do you want to close this in favor of #1124 ?

**Pull request closed without merging**